Reduce CPU usage on the schedd by enabling PER_JOB_HISTORY_DIR. Fixes…

juztas opened the pull request:
… #4618

juztas commented:
@AndresTanasijczuk @bbockelm @mmascher please can you review? I am now testing with turned of PER_JOB_HISTORY_DIR

bbockelm commented on scripts/dag_bootstrap_startup.sh:
```diff
@@ -140,6 +140,16 @@ export _condor_DAGMAN_CONDOR_RM_EXE=$PWD/condor_rm_fix
+# Old logic: Use condor_q -debug -userlog <user_log_file>.
+# Please remember that any condor_q command is expensive to scheduler.
+# See: https://github.com/dmwm/CRABServer/issues/4618
+export _CONDOR_PER_JOB_HISTORY_DIR=/data/srv/glidecondor/condor_local/job_history/
```
Take this from condor_config_val instead?

juztas replied:
changed

bbockelm commented on src/python/TaskWorker/Actions/PostJob.py:
```diff
@@ -1075,15 +1075,41 @@ def execute_internal(self):
+        ## Parse the job ad and use it if possible.
+        ## If not, use main ROOT job ad and for job ad information use condor_q
+        ## Please see: https://github.com/dmwm/CRABServer/issues/4618
+        use_root_job_ad = 1
```
Use true/false instead of 1/0.

bbockelm commented on src/python/TaskWorker/Actions/RetryJob.py:
```diff
@@ -338,7 +338,10 @@ def execute_internal(self, reqname, job_return_code, crab_retry, job_id, dag_job
 
-        self.get_job_ad()
+        if not use_root_job_ad:
+            self.get_job_ad()
```
An important part of get_job_ad() is that it puts all execution attempts into self.ads -- not just the latest one!  This is critical for check_cpu_report.

I think you need to append the latest job ad to finished_jobs/$DAG_NODE_ID.ads, then have get_job_ad parse that file instead of doing its condor_q.

bbockelm commented:
Alright - needs some more work.  The most critical missing piece is that we need to maintain a history of *all* node job executions, not just the last one.

I don't understand any of the use_root_job_ad stuff. Why would we ever use that for RetryJob?

juztas commented:
@bbockelm Please can you review again? I changed almost everything and changed name to used_job_ad and it is still passed to RetryJob.py. I am passing it because if I was able to load job ad in PostJob, I dont see the reason load it again, I dont have preference here, so how you will say :smile:
